### PR TITLE
Move debug stream

### PR DESCRIPTION
**Pull request by marehr**
Left-over of https://github.com/seqan/product_backlog/issues/160

### CODE REVIEW

File: include/seqan3/alignment/matrix/debug_matrix.hpp [hunk 1]
```diff
@@ -17,10 +17,10 @@
...
 #include <seqan3/alignment/matrix/row_wise_matrix.hpp>
 #include <seqan3/alignment/matrix/trace_directions.hpp>
-#include <seqan3/core/detail/debug_stream_alphabet.hpp>
-#include <seqan3/core/detail/debug_stream_range.hpp>
-#include <seqan3/core/detail/debug_stream_type.hpp>
-#include <seqan3/core/detail/debug_stream_optional.hpp>
+#include <seqan3/alphabet/detail/debug_stream_alphabet.hpp>
+#include <seqan3/core/debug_stream/range.hpp>
+#include <seqan3/core/debug_stream/debug_stream_type.hpp>
+#include <seqan3/core/debug_stream/optional.hpp>
```
Comment (MitraDarja) [inline include/seqan3/alignment/matrix/debug_matrix.hpp:23]: Alphabetical order?

File: include/seqan3/alignment/matrix/detail/advanceable_alignment_coordinate.hpp [hunk 2]
```diff
@@ -17,8 +17,8 @@
 #include <type_traits>
 
 #include <seqan3/alignment/matrix/detail/matrix_coordinate.hpp>
-#include <seqan3/core/detail/debug_stream_tuple.hpp>
-#include <seqan3/core/detail/debug_stream_type.hpp>
+#include <seqan3/core/debug_stream/tuple.hpp>
+#include <seqan3/core/debug_stream/debug_stream_type.hpp>
```
Comment (MitraDarja) [inline include/seqan3/alignment/matrix/detail/advanceable_alignment_coordinate.hpp:21]: Alphabetical order?

File: include/seqan3/core/detail/all.hpp [hunk 3]
```diff
@@ -14,12 +14,12 @@
...
-#include <seqan3/core/detail/debug_stream_optional.hpp>
-#include <seqan3/core/detail/debug_stream_range.hpp>
-#include <seqan3/core/detail/debug_stream_tuple.hpp>
-#include <seqan3/core/detail/debug_stream_type.hpp>
-#include <seqan3/core/detail/debug_stream_variant.hpp>
+#include <seqan3/alphabet/detail/debug_stream_alphabet.hpp>
+#include <seqan3/core/debug_stream/optional.hpp>
+#include <seqan3/core/debug_stream/range.hpp>
+#include <seqan3/core/debug_stream/tuple.hpp>
+#include <seqan3/core/debug_stream/debug_stream_type.hpp>
```
Comment (MitraDarja) [inline include/seqan3/core/detail/all.hpp:21]: Alphabetical order?

**Review by MitraDarja — CHANGES_REQUESTED** [review]
LGTM, just some :nail_care:

File: include/seqan3/alignment/matrix/debug_matrix.hpp [hunk 4]
```diff
@@ -17,10 +17,10 @@
 #include <seqan3/alignment/matrix/matrix_concept.hpp>
 #include <seqan3/alignment/matrix/row_wise_matrix.hpp>
 #include <seqan3/alignment/matrix/trace_directions.hpp>
-#include <seqan3/core/detail/debug_stream_alphabet.hpp>
-#include <seqan3/core/detail/debug_stream_range.hpp>
-#include <seqan3/core/detail/debug_stream_type.hpp>
-#include <seqan3/core/detail/debug_stream_optional.hpp>
+#include <seqan3/alphabet/detail/debug_stream_alphabet.hpp>
+#include <seqan3/core/debug_stream/range.hpp>
+#include <seqan3/core/debug_stream/debug_stream_type.hpp>
```
Comment (marehr) [inline include/seqan3/alignment/matrix/debug_matrix.hpp:22]:
```suggestion
#include <seqan3/core/debug_stream/debug_stream_type.hpp>
#include <seqan3/core/debug_stream/range.hpp>
```

File: include/seqan3/alignment/matrix/detail/advanceable_alignment_coordinate.hpp (hunk 2 above)
Comment (marehr) [inline include/seqan3/alignment/matrix/detail/advanceable_alignment_coordinate.hpp:21]:
```suggestion
#include <seqan3/core/debug_stream/debug_stream_type.hpp>
#include <seqan3/core/debug_stream/tuple.hpp>
```

File: include/seqan3/core/detail/all.hpp (hunk 3 above)
Comment (marehr) [inline include/seqan3/core/detail/all.hpp:21]:
```suggestion
#include <seqan3/core/debug_stream/debug_stream_type.hpp>
#include <seqan3/core/debug_stream/optional.hpp>
#include <seqan3/core/debug_stream/range.hpp>
#include <seqan3/core/debug_stream/tuple.hpp>
```

File: include/seqan3/alignment/matrix/debug_matrix.hpp (hunk 1 above)
Comment (marehr) [inline include/seqan3/alignment/matrix/debug_matrix.hpp:23]:
```suggestion
#include <seqan3/alphabet/detail/debug_stream_alphabet.hpp>
#include <seqan3/core/debug_stream/debug_stream_type.hpp>
#include <seqan3/core/debug_stream/optional.hpp>
#include <seqan3/core/debug_stream/range.hpp>
```

**Review by MitraDarja — APPROVED** [review]

**Review by eseiler — APPROVED** [review]

**Pull request merged**